### PR TITLE
Add Google Cloud Storage example

### PR DESCRIPTION
**Pull request by dstansby**
This adds an example of how to open an array that's in Google Cloud Storage. Fixes https://github.com/zarr-developers/zarr-python/issues/1684

I've added a new section to the docs, "Examples", for short, self-contained examples, that achieve a specific goal. This is distinct from existing sections of the docs, and should complement the existing tutorial and reference sections. For those familiar with [Diataxis](https://diataxis.fr/), this is essentially the "how-to guides" quadrant.

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

**Comment by d-v-b** [comment]
Your example uses `N5FSStore`, perhaps an example for a wide audience should just use `FSStore`?

File: docs/examples/google_cloud.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    import zarr
+    import gcsfs
+
+    project = "my_project"
+    bucket = "my_bucket"
+    path_to_array = "a/b/c"
+
+    fs = gcsfs.GCSFileSystem(project=bucket, token='anon', access='read_only')
+    store = zarr.N5FSStore(url=bucket, fs=fs)
```
Comment (dstansby) [inline docs/examples/google_cloud.rst:17]:
```suggestion
    store = zarr.FSStore(url=bucket, fs=fs)
```

File: docs/examples/google_cloud.rst [hunk 2]
```diff
@@ -0,0 +1,18 @@
+Opening from Goolge Cloud Storage
```
Comment (MSanKeys963) [inline docs/examples/google_cloud.rst:1]:
```suggestion
Opening from Google Cloud Storage
```

File: docs/examples/google_cloud.rst [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    import zarr
+    import gcsfs
+
+    project = "my_project"
+    bucket = "my_bucket"
+    path_to_array = "a/b/c"
+
+    fs = gcsfs.GCSFileSystem(project=bucket, token='anon', access='read_only')
+    store = zarr.FSStore(url=bucket, fs=fs)
```
Comment (MSanKeys963) [inline docs/examples/google_cloud.rst:17]: This line throws an error:

```python
AttributeError: module 'zarr' has no attribute 'FSStore'
```

**Comment by MSanKeys963** [comment]
Hi @dstansby, thanks for sending the PR.

I wonder if this example should be listed [here](https://zarr.readthedocs.io/en/stable/tutorial.html#distributed-cloud-storage) along with S3 and Azure examples?

**Pull request closed without merging**